shared/middleware/helpers/redis_table.rb

```diff
@@ -0,0 +1,108 @@
+# RedisTable provides a table abstraction that notifies clients of changes.
+# Operations include adding a row, updating and fetching a specific row
+# by id, and fetching all rows. Rows preserve the order of insertion.
+# # The implementation wraps a RedisProperty bag to store the table rows
+# # and a pub sub service to notify clients of changes.
+
+require_relative 'redis_property_bag'
+
+# @private
+class RedisTable
+  ROW_ID_KEY = "row_id"
+
+  def initialize(redis, pub_sub_api, shard_id, table_name)
```

Comment:
Should it be safe to create a RedisTable without a pub_sub_api, in case we never want certain tables to publish?  For example, when we do the Pusher presence integration we'll need to keep a mapping of Pusher-user-id to node-id and this would be a nice way to do that, but we never need to tell clients about changes to that table.